frontend/packages/helm-plugin/src/components/list-page/HelmTabbedPage.tsx

```diff
@@ -0,0 +1,122 @@
+import CreateProjectListPage from '@console/dev-console/src/components/projects/CreateProjectListPage';
+import { useAccessReview } from '@console/dynamic-plugin-sdk/src';
+import { withStartGuide } from '@console/internal/components/start-guide';
+import { Page } from '@console/internal/components/utils';
+import { MenuActions, MultiTabListPage } from '@console/shared';
+import { HelmChartRepositoryModel, ProjectHelmChartRepositoryModel } from '../../models';
+import HelmReleaseList from './HelmReleaseList';
+import HelmReleaseListPage from './HelmReleaseListPage';
+import RepositoriesPage from './RepositoriesListPage';
+
+type HelmTabbedPageProps = RouteComponentProps<{ ns: string }>;
+
+export const PageContents: React.FC<HelmTabbedPageProps> = (props) => {
+  const { t } = useTranslation();
+  const {
+    match: {
+      params: { ns: namespace },
+    },
+  } = props;
+  const [showTitle, canCreate] = [false, false];
+  const [projectHelmChartCreateAccess, loadingCreatePHCR] = useAccessReview({
+    group: ProjectHelmChartRepositoryModel.apiGroup,
+    resource: ProjectHelmChartRepositoryModel.plural,
+    verb: 'create',
+    namespace,
+  });
+  const [helmChartCreateAccess, loadingCreateHCR] = useAccessReview({
+    group: HelmChartRepositoryModel.apiGroup,
+    resource: HelmChartRepositoryModel.plural,
+    verb: 'create',
+  });
+  const [projectHelmChartEditAccess, loadingEditPHCR] = useAccessReview({
+    group: ProjectHelmChartRepositoryModel.apiGroup,
+    resource: ProjectHelmChartRepositoryModel.plural,
+    verb: 'update',
+    namespace,
+  });
+  const [helmChartEditAccess, loadingEditHCR] = useAccessReview({
+    group: HelmChartRepositoryModel.apiGroup,
+    resource: HelmChartRepositoryModel.plural,
+    verb: 'update',
+  });
+
+  const menuActions: MenuActions = {
+    helmRelease: {
+      label: t('helm-plugin~Helm Release'),
+      onSelection: () => `/catalog/ns/${namespace}?catalogType=HelmChart`,
+    },
+    projectHelmChartRepository: {
+      label:
+        projectHelmChartCreateAccess || helmChartCreateAccess ? t('helm-plugin~Repository') : null,
+      onSelection: () => `/ns/${namespace}/helmchartrepositories/~new?actionOrigin=list`,
+    },
+  };
+
+  const pages: Page[] = [
+    {
+      href: '',
+      name: t('helm-plugin~Helm Releases'),
+      component: HelmReleaseList,
+    },
+    {
+      href: 'repositories',
+      name: t('helm-plugin~Repositories'),
+      component: RepositoriesPage,
+      pageData: {
+        showTitle,
+        canCreate,
+      },
+    },
+  ];
+
+  return namespace ? (
+    projectHelmChartCreateAccess ||
+    projectHelmChartEditAccess ||
+    helmChartCreateAccess ||
+    helmChartEditAccess ? (
+      <MultiTabListPage
+        pages={pages}
+        match={props.match}
+        title={t('helm-plugin~Helm')}
+        menuActions={menuActions}
+      />
+    ) : (
+      !loadingCreatePHCR &&
+      !loadingEditPHCR &&
+      !loadingCreateHCR &&
```

Comment:
could you explain why is this required?

Reply:
Since we are making call to check access for user, till we get response it will not render the component. If I don't add this, user may experience a glitch if the network is slow. This is just to make sure that till we get response from useAccessReview don't render any component.